tests/kernel/smp/src/main.c

```diff
@@ -14,15 +13,31 @@
 #error SMP test requires at least two CPUs!
 #endif
 
-#define T2_STACK_SIZE 2048
-
-K_THREAD_STACK_DEFINE(t2_stack, T2_STACK_SIZE);
+#define THREAD_DELAY 200
+#define DELAY_US 50000
+#define STACK_SIZE (384 + CONFIG_TEST_EXTRA_STACKSIZE)
+#define THREADS_NUM 10
 
+K_THREAD_STACK_DEFINE(t2_stack, STACK_SIZE);
 struct k_thread t2;
-
 volatile int t2_count;
 
-#define DELAY_US 50000
+static K_THREAD_STACK_ARRAY_DEFINE(tstack, THREADS_NUM, STACK_SIZE);
+static struct k_thread tthread[THREADS_NUM];
+
+struct tthread_data {
+	k_tid_t tid;
+	int priority;
+	int executed;
+};
+
+static struct tthread_data tdata[THREADS_NUM];
```

Comment:
If you want to test these are getting set from different threads, you generally want to flag the array volatile to prevent the compiler from messing things up.  See t2_count above.